aries-site/src/examples/templates/filtering/FilteringLists.js

```diff
@@ -0,0 +1,251 @@
+    locationType: 'Office',
+    name: 'Madrid, Spain (ESM)',
+    status: 'Offline',
+  },
+  {
+    address:
+      '3404 E Harmony Road, Fort Collins, Colorado, 80528, United States',
+    image:
+      'https://images.unsplash.com/photo-1503424886307-b090341d25d1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1955&q=80',
+    locationType: 'Office',
+    name: 'Fort Collins, CO',
+    status: 'Online',
+  },
+  {
+    address:
+      '6280 America Center Dr., San Jose, California, 95002, United States',
+    image:
+      'https://images.unsplash.com/photo-1501594907352-04cda38ebc29?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2389&q=80',
+    locationType: 'Customer Center',
+    name: 'WW Corporate Headquarters - San Jose, CA',
+    status: 'Online',
+  },
+  {
+    address: '461 Fifth Avenue, New York, NY, 10017, United States',
+    image:
+      'https://images.unsplash.com/photo-1534430480872-3498386e7856?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80',
+    locationType: 'Customer Center',
+    name: 'New York, NY',
+    status: 'Offline',
+  },
+  {
+    address: `Stroombaan 16, 1181 VX Amstelveen, 
+    Amstelveen, The Netherlands`,
+    image:
+      'https://images.unsplash.com/photo-1584450150050-4b9bdbd51f68?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80',
+    locationType: 'Customer Center',
+    name: 'Amstelveen HQ, NL',
+    status: 'Online',
+  },
+];
+
+const defaultFilters = {};
+const defaultLocationType = 'All Location Types';
+const defaultStatus = 'All statuses';
+
+export const FilteringLists = () => {
+  const [data, setData] = useState(allData);
+  const [filtering, setFiltering] = useState(false);
+  const [filters, setFilters] = useState(defaultFilters);
+  const [locationType, setLocationType] = useState(defaultLocationType);
+  const [status, setStatus] = useState(defaultStatus);
+  const size = useContext(ResponsiveContext);
+
+  const resetFilters = () => {
+    setData(allData);
+    setStatus(defaultStatus);
+    setLocationType(defaultLocationType);
+    setFiltering(false);
+    setFilters(defaultFilters);
+  };
+
+  useEffect(() => {
+    if (locationType === defaultLocationType && status === defaultStatus) {
+      resetFilters();
+    } else {
+      setFiltering(true);
+    }
+  }, [locationType, status]);
+
+  const filterData = (array, criteria) => {
+    setFilters(criteria);
+    const filterKeys = Object.keys(criteria);
+    return array.filter(item => {
+      // validates all filter criteria
+      return filterKeys.every(key => {
+        // ignores non-function predicates
+        if (typeof criteria[key] !== 'function') return true;
+        return criteria[key](item[key]);
+      });
+    });
+  };
+
+  return (
+    <Box
+      fill
+      gap="medium"
+      margin="auto"
+      overflow="auto"
+      pad="medium"
+      width={{ max: 'xxlarge' }}
+    >
+      <Header>
```

Comment:
Similar to how we introduced the concept of `<Identity />` I'd like to explore the idea of a `<FilterControls />`. Can we extract this into a focussed component?

Probably would pull out Heading.

So would have:
<Heading />
<FilterControls />
<ResultSet /> or <Sites />

Reply:
Yes definitely, I've actually started this locally but just hadn't cleaned up the examples. Glad we're thinking the same direction.